flang/lib/Semantics/resolve-names.cpp

```diff
@@ -1605,6 +1609,35 @@ void OmpVisitor::Post(const parser::OpenMPBlockConstruct &x) {
   }
 }
 
+// This "manually" walks the tree of the cosntruct, because the order
+// elements are resolved by the normal visitor will try to resolve
+// the map clauses attached to the directive without having resolved
+// the type, so the type is figured out using the implicit rules.
+bool OmpVisitor::Pre(const parser::OpenMPDeclareMapperConstruct &x) {
+  AddOmpSourceRange(x.source);
+  BeginDeclTypeSpec();
+  const auto &spec{std::get<parser::OmpDeclareMapperSpecifier>(x.t)};
+  Symbol *mapperSym{nullptr};
+  if (const auto &mapperName{std::get<std::optional<parser::Name>>(spec.t)}) {
+    mapperSym =
+        &MakeSymbol(*mapperName, MiscDetails{MiscDetails::Kind::ConstructName});
+    mapperName->symbol = mapperSym;
+  } else {
+    mapperSym = &MakeSymbol(
+        "default", Attrs{}, MiscDetails{MiscDetails::Kind::ConstructName});
```

Comment:
To be honest, not sure. I'm not sure how the OTHER symbol is retrieved either. 

@TIFitis Any thoughts?

Reply:
Intuitively I think it would be possible to do so.

I am working on further patches for this to have full lowering support, we can make/add changes to the parsing if needed as part of those patches I guess.

Reply:
In the `if` case, we can get the name from the parse-tree node and get the symbol. But we can wait for @TIFitis's lowering patch to see whether it can be used. Either way, adding the symbol seems to have caught the duplicate `default` mapper for the same type that @TIFitis raised in another comment.